js_modules/dagit/packages/core/src/instigation/TickHistory.tsx

```diff
@@ -360,6 +377,56 @@ export const TickHistoryTimeline = ({
   );
 };
 
+function DynamicPartitionRequestsCell({
+  requests,
+}: {
+  requests: DynamicPartitionsRequestResultFragment[];
+}) {
+  const [isDialogOpen, setDialogOpen] = React.useState(false);
+  if (!requests.some((request) => request.partitionKeys?.length)) {
+    return null;
+  }
+
+  return (
+    <>
+      <ButtonLink
+        onClick={() => {
+          setDialogOpen(true);
+        }}
+      >
+        {requests.length} dynamic partition change{requests.length === 1 ? '' : 's'}
+      </ButtonLink>
+      <Dialog
+        isOpen={isDialogOpen}
+        onClose={() => {
+          setDialogOpen(false);
+        }}
+        style={{width: '60%', minWidth: '400px'}}
+        title={
+          <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+            <Icon name="partition" />
+            Dynamic partition changes
+          </Box>
+        }
+      >
+        <DialogBody>
```

Comment:
It would probably look a little cleaner to remove the `DialogBody` to get the table edges flush with the edges of the dialog.